pennylane_lightning/src/simulators/lightning_kokkos/gates/GateFunctors.hpp

```diff
@@ -0,0 +1,3164 @@
+#pragma once
```

Comment:
At a later time, we can probably break this file into parametric and non-parametric gates to help with readability.

Reply:
I'll split it, it will be more readable indeed.